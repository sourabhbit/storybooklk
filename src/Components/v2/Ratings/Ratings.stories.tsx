import { ComponentMeta, ComponentStory } from '@storybook/react'
import Ratings from './index'
import { ComponentSizeENUM } from 'Types/general'
import { generateArgTypes } from 'Helpers/Storybook'
// import { iconsENUM } from '../../Types/icons'

export default {
	title: 'Components/v2/Rating/Ratings',
	component: Ratings,
	argTypes: {
		id: generateArgTypes({
			type: 'string',
			required: true,
			description: 'String used as unique identifier',
			category: 'General',
		}),
		componentSize: generateArgTypes({
			type: 'string',
			required: false,
			description: 'Size of the component',
			category: 'General',
			defaultValue: ComponentSizeENUM.small,
			control: 'radio',
			options: ComponentSizeENUM,
		}),
		rating: generateArgTypes({
			type: 'number',
			required: true,
			description: 'Rating value',
			category: 'Component',
		}),
		max: generateArgTypes({
			type: 'number',
			required: true,
			description: 'Max allowed rating',
			category: 'Component',
		}),
		icon: generateArgTypes({
			type: 'string',
			required: false,
			description: 'Rating Icon',
			category: 'Component',
			control: 'select',
			defaultValue: 'starFilled',
			// options: iconsENUM,
		}),
	},
} as ComponentMeta<typeof Ratings>

const Template: ComponentStory<typeof Ratings> = (args) => <Ratings {...args} />

export const Overview = Template.bind({})
Overview.args = {
	id: 'ratings-stars',
	rating: 3,
	max: 5,
	componentSize: ComponentSizeENUM.medium,
	// icon: iconsENUM.starFilled,
}
