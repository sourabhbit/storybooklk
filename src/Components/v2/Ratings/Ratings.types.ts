import { ComponentSizeENUM } from 'Types/general'
import { iconsENUM } from 'Types/icons'
export interface RatingsTypes {
	rating: number
	max: number
	componentSize?: ComponentSizeENUM
	id: string
	icon?: iconsENUM
}
