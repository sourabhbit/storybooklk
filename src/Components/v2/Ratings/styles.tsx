import styled from 'styled-components'
import { ComponentSizeENUM } from 'Types/general'

export const RatingsWrapper = styled.div<{
	styleSize: ComponentSizeENUM
}>`
	display: flex;
	overflow: hidden;
`
export const SVGComponentWrapper = styled.div<{
	styleSize: ComponentSizeENUM
}>`
	margin-right: var(--spacing-sm);
	height: ${(props) =>
		props.styleSize === ComponentSizeENUM.small
			? '20px'
			: props.styleSize === ComponentSizeENUM.medium
			? '40px'
			: props.styleSize === ComponentSizeENUM.large
			? '60px'
			: ''};

	width: ${(props) =>
		props.styleSize === ComponentSizeENUM.small
			? '20px'
			: props.styleSize === ComponentSizeENUM.medium
			? '40px'
			: props.styleSize === ComponentSizeENUM.large
			? '60px'
			: ''};
`
