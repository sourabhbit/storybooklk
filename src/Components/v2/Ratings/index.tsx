import { ComponentSizeENUM } from 'Types/general'
import { iconsENUM } from 'Types/icons'
import SVGIcon from 'Components/common/SVGIcon'
import { RatingsTypes } from './Ratings.types'
import { RatingsWrapper, SVGComponentWrapper } from './styles'
const Ratings = ({
	componentSize = ComponentSizeENUM.medium,
	rating = 3.5,
	max = 5,
	icon,
}: RatingsTypes) => {
	return (
		<RatingsWrapper styleSize={componentSize} data-testid="ratings-test">
			{Array.from({ length: max }, (elem, idx) => {
				{
					icon =
						idx + 1 <= Math.floor(rating)
							? iconsENUM.starFilled
							: iconsENUM.star
				}
				return (
					<SVGComponentWrapper styleSize={componentSize} key={idx}>
						{icon && <SVGIcon icon={icon} alt="star" />}
					</SVGComponentWrapper>
				)
			})}
		</RatingsWrapper>
	)
}

export default Ratings
