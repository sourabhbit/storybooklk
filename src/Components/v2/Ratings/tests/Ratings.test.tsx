import { render, screen } from '@testing-library/react'
import Ratings from '../index'

const props = {
	rating: 4.5,
	max: 5,
}

describe('<Ratings />', () => {
	let wrapper: any = null

	beforeEach(() => {
		const { asFragment } = render(<Ratings {...props} />)
		wrapper = asFragment
	})

	afterAll(() => {
		wrapper = null
	})

	it('Should take a snapshot', () => {
		expect(wrapper()).toMatchSnapshot()
	})
	it('Should render Ratings component', () => {
		const container = screen.getByTestId('ratings-test')
		expect(container).toBeInTheDocument()
	})
})
