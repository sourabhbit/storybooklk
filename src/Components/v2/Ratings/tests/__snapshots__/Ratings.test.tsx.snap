// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ratings /> Should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL gDgBQA"
    data-testid="ratings-test"
  >
    <div
      class="sc-gsDKAQ kxoApa"
    >
      <img
        alt="star"
        src="/starFilled.svg"
      />
    </div>
    <div
      class="sc-gsDKAQ kxoApa"
    >
      <img
        alt="star"
        src="/starFilled.svg"
      />
    </div>
    <div
      class="sc-gsDKAQ kxoApa"
    >
      <img
        alt="star"
        src="/starFilled.svg"
      />
    </div>
    <div
      class="sc-gsDKAQ kxoApa"
    >
      <img
        alt="star"
        src="/starFilled.svg"
      />
    </div>
    <div
      class="sc-gsDKAQ kxoApa"
    >
      <img
        alt="star"
        src="/star.svg"
      />
    </div>
  </div>
</DocumentFragment>
`;
